lib/rdoc/generator/darkfish.rb

```diff
@@ -824,6 +824,50 @@ def generate_ancestor_list(ancestors, klass)
     content << '</li></ul>'
   end
 
+  def generate_class_link(klass, rel_prefix)
+    if klass.display?
+      %(<code><a href="#{rel_prefix}/#{klass.path}">#{klass.name}</a></code>)
+    else
+      %(<code>#{klass.name}</code>)
+    end
+  end
+
+  def generate_class_index_content(classes, rel_prefix)
+    grouped_classes = group_classes_by_namespace_for_sidebar(classes)
+    return '' unless top = grouped_classes[nil]
+
+    solo = top.one? { |klass| klass.display? }
+    traverse_classes(top, grouped_classes, rel_prefix, solo)
+  end
+
+  def traverse_classes(klasses, grouped_classes, rel_prefix, solo = false)
+    content = +'<ul class="link-list">'
+
+    klasses.each do |index_klass|
+      if children = grouped_classes[index_klass.full_name]
+        content << %(<li><details#{solo ? ' open' : ''}><summary>#{generate_class_link(index_klass, rel_prefix)}</summary>)
+        content << traverse_classes(children, grouped_classes, rel_prefix)
+        content << '</ul></details>'
+        solo = false
+      elsif index_klass.display?
+        content << %(<li>#{generate_class_link(index_klass, rel_prefix)})
+      end
+    end
+
+    content
```

Comment:
Not actually related to this pull request change, but can we add a closing `</ul>` here, and closing `</li>` in `klasses.each`?
Closing `</li>` tag can be omitted but it would be much clear if the opening and closing tag corresponds.
Closing `</ul>` is mandatory (https://developer.mozilla.org/en-US/docs/Web/HTML/Element/ul#technical_summary) and the html created by rdoc master seems invalid.

Reply:
Good catch. Updated 👍